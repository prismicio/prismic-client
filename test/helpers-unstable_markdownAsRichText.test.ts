import { describe } from "vitest";

import { testMarkdownAsRichTextHelper } from "./__testutils__/testAsRichTextHelper";

describe("transforms markdown to rich text", () => {
	describe("basic", () => {
		testMarkdownAsRichTextHelper("empty", {
			input: /* md */ ``,
		});

		testMarkdownAsRichTextHelper("single tag", {
			input: /* md */ `lorem ipsum dolor sit amet`,
		});

		testMarkdownAsRichTextHelper("multiple tags", {
			input: /* md */ `# lorem ipsum dolor sit amet
			consectetur adipiscing elit`,
		});
	});
});
