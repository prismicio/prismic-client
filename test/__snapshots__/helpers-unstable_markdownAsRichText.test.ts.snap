// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transforms markdown to rich text > basic > empty 1`] = `[]`;

exports[`transforms markdown to rich text > basic > multiple tags 1`] = `
[
  {
    "direction": "ltr",
    "spans": [],
    "text": "lorem ipsum dolor sit amet",
    "type": "heading1",
  },
  {
    "direction": "ltr",
    "spans": [],
    "text": "consectetur adipiscing elit",
    "type": "preformatted",
  },
]
`;

exports[`transforms markdown to rich text > basic > single tag 1`] = `
[
  {
    "direction": "ltr",
    "spans": [],
    "text": "lorem ipsum dolor sit amet",
    "type": "paragraph",
  },
]
`;
